ec2-cluster/modules/node/main.tf

```diff
@@ -0,0 +1,198 @@
+resource "aws_instance" "node" {
+  count = var.instance_count
+
+  ami                         = data.aws_ami.node.id
+  instance_type               = var.instance_type
+  associate_public_ip_address = true
+
+  root_block_device {
+    volume_type = "gp2"
+    volume_size = var.disk_size
+  }
+
+  user_data = file("ssh_keys.sh")
+
+  tags = {
+    Index     = count.index
+    Name      = "node-${var.cluster_id}-${count.index}"
+    Env       = var.env
+    Cluster   = var.cluster_id
+    Workspace = var.workspace
+  }
+
+  connection {
+    host    = self.public_ip
+    type    = "ssh"
+    user    = local.ssh_user
+    timeout = "240s"
+  }
+
+  provisioner "file" {
+    content = templatefile("${path.module}/templates/setup", {
+      tessellation_version = var.tessellation_version,
+      key                  = var.instance_keys[count.index].key
+      app_env              = var.env
+    })
+    destination = "/tmp/setup"
+  }
+
+  provisioner "file" {
+    source      = "${path.module}/templates/setup-services"
+    destination = "/tmp/setup-services"
+  }
+
+  provisioner "file" {
+    source      = "${path.module}/data/key-${count.index}.p12"
+    destination = "/tmp/key.p12"
+  }
+
+  provisioner "file" {
+    content = templatefile("${path.module}/templates/rollback", {
+      public_ip            = self.public_ip,
+      app_env              = var.env,
+      public_port          = var.public_port,
+      p2p_port             = var.p2p_port,
+      cli_port             = var.cli_port,
+      snapshot_stored_path = var.snapshot_stored_path
```

Comment:
Let's reduce the code duplication by storing this duplicated object in the `local` variable and just refer to it in the `templatefile` function.

Reply:
https://github.com/Constellation-Labs/infrastructure_utils/pull/22/files